model/project_ref.go

```diff
@@ -2244,55 +2260,52 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
 	return res, nil
 }
 
-// ValidateProjectCreation returns a boolean if you should surface the error or not.
-func ValidateProjectCreation(projectId string, config *evergreen.Settings, projectRef *ProjectRef) (bool, error) {
+// ValidateEnabledProjectsLimit takes in a merged project ref and validates if the current project is allowed to be enabled.
+// Returns a status code and error we are already at limit with enabled projects.
+func ValidateEnabledProjectsLimit(projectId string, config *evergreen.Settings, mergedRefToValidate ProjectRef) (int, error) {
```

Comment:
Preference for mergedRefToValidate to be a pointer since that's better for passing -- we don't need to check if it's nil if that's not correct to pass in.